woocommerce/woocommerce-gateway-stripe
# Fix payment method description on checkout page

Fixes #3272 
Some more discussions in p1720692041109859-slack-C055WHLA98D

If the description is set to empty, we are not showing the default description on the checkout page now.

## Changes proposed in this Pull Request:
- When the description of a payment method is empty, we are now returning empty instead of the default description to show on the checkout page ([8af6045](https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3282/commits/8af6045f95a05c495f34ccb617fa050fc72171d4)). 
- This PR has two more additional tweak commits.
    - Removed this unnecessary response data https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3282/commits/b70d6f7cdaa59d9acbb9fd4fc2d04d5a49cf5b22
    - Removed this Multibanco specific condition which was needed when Multibanco was available only in Sources API regardless of UPE enabled/disabled (https://github.com/woocommerce/woocommerce-gateway-stripe/pull/3282/commits/f7f1a061fdfcb85b96043a8d3737ad7fff82e5e5). As Multibanco is available is UPE now, this check is unnecessary. 

## Testing instructions
- Go to the Stripe settings page and disable the legacy checkout experience (enable UPE).
- Go to the payment methods tab.
- Enable the card method and make the description empty.
- Go to the checkout page and check the card element.
- In `develop` branch you would see the admin facing default description showing up there. In this branch no default description will show up on the checkout page when the description is empty.
- Test this for other Stripe payment methods too. 
- Test the legacy and new checkout experience. 
- Test on shortcode and block checkout page.
